From 8679532a75775f671cd3b043c722329791542ec5 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 12 Oct 2016 00:46:16 +0100
Subject: [PATCH 8/8] dependencies: Optimize lookup of external symbols in
 packages

It makes absolutely no sense to search for paths that don't exist, so to
drastically speed up the computation time we skip all bound paths that
don't exist in our local filesystem.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/dependencies.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ypkg2/dependencies.py b/ypkg2/dependencies.py
index 6f59e7a..f182713 100644
--- a/ypkg2/dependencies.py
+++ b/ypkg2/dependencies.py
@@ -111,6 +111,8 @@ class DependencyResolver:
         pkg = None
         for path in paths:
             fpath = os.path.join(path, symbol)
+            if not os.path.exists(fpath):
+                continue
             lpkg = None
             if fpath in self.files_cache:
                 lpkg = self.files_cache[fpath]
-- 
2.10.1

